### DISCUSSION TITLE
So AlDente will stop my M1 Mac from sleeping, is it bad?

### DISCUSSION THREAD

**Discussion by teitoku1006** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#132] [Q&A]
Hello dev, English is not my native language so it might be hard for you to understand so sorry in advanced. I have a question about using  AlDente on M1 Mac. 

So from what I understand, AlDente will stop my M1 Mac from charging more than the level I set it as (Example: set at 50%, battery might charge to 53 => 55% and then stop). 

1) BUT it also said in the app is " Prevent Sleep: true ", does that mean my M1 Macbook Air will always be turn on and running (With the screen off because of inactivity) and not able to sleep? is always turn on like this bad for my Mac?

1.5) So I turn off the " Wake for network access " and in AlDente it changed Prevent Sleep from true to false. But When I reactive the " Wake for network access " AlDente still said " false ". Possible bug?

2) If it's better for my MBA M1 to sleep when not using, is it possible for my MBA M1 Mac to sleep and the battery will not be charging more than the level I set it stopping at? (Example like 50%)

3) If I plug in my MBA M1 Mac and shut down the machine, will the battery keep charging because AlDente is not running anymore?

**Comment by LeshawnG** [comment]
Hello, I will try to answer these questions best based on my experience with the MBP M1 and AlDente.

You're correct, AlDente will stop/inhibit charging at the set level. If AlDente is set to 50%, the MB battery will charge until 50% then stop and when charging is stopped, it runs on AC power. It should appear like this.
<img width="299" alt="sc" src="https://user-images.githubusercontent.com/73793035/108633989-bdc46280-744d-11eb-866d-d29d294f7ad4.png">
However, you mentioned it might charge to 53 => 55% then stop but I do not experience this. Charging stops at the set level which is 50% in my case. 

1. From my use, Prevent Sleep is only true while the MB is charging. When the set charge level is reached, Prevent Sleep is changed to false. The MB will sleep like normal when the set charge level is reached and it does not charge while the asleep. v2.0 was charging while awake and asleep but v2.1 has fixed this and no longer charges. I can confirm no charging while asleep and the MB will sleep when set charge level is reached.

1.5. I am unsure about wake for network access but I do have mine enabled. I'm wondering if this can be the reason why my MB is not charging while asleep as it periodically wakes which allows AlDente to keep working.

2. Correct, it's better to sleep when not using and the battery will not charge while asleep.

3. I am unsure about charging while powered off as I have not tried it myself. However, since the SMC is essentially reset during power off and AlDente is not running, I would expect the MB to charge while asleep.

**Reply by teitoku1006** [reply]
Such detail reply!

I thank you for answer all of my questions, I'm no longer worry now

I have 2 more questions:

1) 
<img width="595" alt="Screen Shot 2021-02-22 at 2 38 09 AM" src="https://user-images.githubusercontent.com/65445582/108636326-1aa83300-74b7-11eb-91dc-2eba13479063.png">

This Picture said that my Power Source is Power Adapter and the Batter is not charging BUT when I go System Preferences => Battery => It said my Battery is Charging and stay still at 55%. (I set at 55% percent in AlDente)
<img width="659" alt="Screen Shot 2021-02-22 at 2 38 20 AM" src="https://user-images.githubusercontent.com/65445582/108636383-7672bc00-74b7-11eb-968a-76d0d0a88b57.png">

Should I be worry? Is my battery already stop charging?  Also, if battery drop to 54%, will it again charge back to 55% again?

2) The AlDente icon on the taskbar is a bit further away from the rest of other icon, is it possible to make it " closer " like the rest?

<img width="595" alt="Screen Shot 2021-02-22 at 2 38 09 AM" src="https://user-images.githubusercontent.com/65445582/108636394-8ee2d680-74b7-11eb-85e0-6ae33999feac.png">

**Reply by LeshawnG** [reply]
You're welcome, trying to help as much as I can!

1.  I believe the reason it's showing in system preferences as "charging" under the battery section is to give you an idea of when the charger is connected.  You see how the whole graph becomes shaded in green when the charger is connected? I believe it's there to give an idea of when the charger is connected and doesn't distinguish from "charging" and "ac power". I'm pretty positive that the MB is using AC power and not charging when the menubar icon displays the battery and plug like you have there. I would not worry. It can simply be that Apple did not think that far to update system preferences to display something like "using AC power" instead of leaving it as charging. 
This is a good observation and I will for sure keep on checking to make sure my of assumption that _"system preferences displays charging while using AC_" still means that the "_MB is not charging._"
I would not be worried and I'm positive that no charging is occuring when the menubar says "battery is not charging"

Additionally, if it drops to 54% then the MB should charge to 55% and stay there. However, I have not seen a 1% drop happen yet while using AlDente. I recommend using coconutBattery or Better Battery 2 as it shows the exact percentage of your battery. You can see I have it next to the macOS battery icon and the real battery percentage is 48.5%. It's much easier to notice a 0.1% increase or decrease using these apps!


2.  Yes it's really simple to rearrange the menubar icons! Press the command key while dragging icons in the menubar to rearrange them

**Reply by teitoku1006** [reply]
Again, thank you for the reply

1) I notice it because before I used Aldente, I turned on the Optimized Charging mode and I noticed that my Battery in System Prefer will said something like: Battery not charging - rarely use in battery mode. 

When using the Apple Optimized Charging mode, The battery in the chart seems to stop charging at 80% but in the menubar will still move up to ~90%, kinda weird. But even then it will still say in Battery (System Prefer) that my Battery not charging - rarely use in battery mode. 

2) Oh thank you for the advice about the menubar, it seems TG Pro it's the culprit making icon far away from each other!

I will also keep paying attention to how my MBA M1 works, thank you!

**Reply by vassudanagunta** [reply]
My understanding is to not use `Apple Optimized Charging` mode if you are using AlDente. You can search Issues for "Apple Optimized Charging" to find the discussion.

**Comment by Limeybastid** [comment]
My actual Big Sur level always shows about 2 % more than what my AL Dente is set to, I believe this to be normal. Example, Al Dente set to 59, actual level on Big Sur stays on 61%.

**Reply by MatthiasKerbl** [reply]
Hi @Limeybastid,
this is completely normal since battery percentages are always a little bit inconsistent.
Best regards,
Matthias

**Comment by fxue1981** [comment]
I'm actually experiencing a different behavior on my M1 MBP. After installing and setting Al dente at 60%, my laptop will show 100% when I plugged in the power cord, when the true battery level is at 95%. Is this expected?
